app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,19 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rule_groups_for_rules = RuleGroup.where(id: rules.includes(:rule_group)
+                                     .references(:rule_group).flat_map(&:rule_group_ancestor_ids))
+
+    conflicts = relationships_for(rules, rule_groups_for_rules, 'conflicts').distinct.map(&:right)
+    requires = relationships_for(rules, rule_groups_for_rules, 'requires').distinct.map(&:right)
```

Comment:
Move `distinct` inside, just keep the `map`

Reply:
I just tried that but forgot that we actually don't want only distinct relationships when we use `relationships_for` for `rules_and_rule_groups`  so I can't add the distinct in that method. I could either just leave distinct where it is or create a `distinct_relationships_for` method for this situation?

Reply:
Why is it a problem to have the `distinct` in the `relationships_for`?

Reply:
I actually was able to update `valid_relationships?` so that it works with duplicates in requires and conflicts so we don't need `distinct` anymore. I added tests to cover duplicate requires/conflicts to make sure it works correctly too.